packages/editor-ui/src/composables/useCanvasOperations.ts

```diff
@@ -1230,11 +1231,85 @@ export function useCanvasOperations({ router }: { router: ReturnType<typeof useR
 		});
 	}
 
+	function revalidateNodeInputConnections(id: string) {
+		const targetNode = workflowsStore.getNodeById(id);
+		if (!targetNode) {
+			return;
+		}
+
+		const nodeType = nodeTypesStore.getNodeType(targetNode.type, targetNode.typeVersion);
+		if (!nodeType) {
+			return;
+		}
+
+		const connections = mapLegacyConnectionsToCanvasConnections(
+			workflowsStore.workflow.connections,
+			workflowsStore.workflow.nodes,
+		);
+
+		connections.forEach((connection) => {
+			if (connection.target === id) {
+				const sourceNode = workflowsStore.getNodeById(connection.source);
+				if (!sourceNode || !connection.data) {
+					return;
+				}
+
+				if (
+					!isConnectionAllowed(
+						sourceNode,
+						targetNode,
+						connection.data.source,
+						connection.data.target,
+					)
+				) {
+					void nextTick(() => deleteConnection(connection));
+				}
+			}
+		});
+	}
+
+	function revalidateNodeOutputConnections(id: string) {
+		const sourceNode = workflowsStore.getNodeById(id);
+		if (!sourceNode) {
+			return;
+		}
+
+		const nodeType = nodeTypesStore.getNodeType(sourceNode.type, sourceNode.typeVersion);
+		if (!nodeType) {
+			return;
+		}
+
+		const connections = mapLegacyConnectionsToCanvasConnections(
+			workflowsStore.workflow.connections,
+			workflowsStore.workflow.nodes,
+		);
+
+		connections.forEach((connection) => {
+			if (connection.source === id) {
+				const targetNode = workflowsStore.getNodeById(connection.target);
+				if (!targetNode || !connection.data) {
+					return;
+				}
+
+				if (
+					!isConnectionAllowed(
+						targetNode,
+						sourceNode,
+						connection.data.source,
+						connection.data.target,
+					)
+				) {
+					void nextTick(() => deleteConnection(connection));
+				}
+			}
+		});
+	}
+
```

Comment:
These two methods are pretty much identical so we could keep it DRY:
```suggestion
function revalidateNodeConnections(id: string, connectionType: 'input' | 'output') {
		const node = workflowsStore.getNodeById(id);
		const isInput = connectionType === 'input';
		if (!node) {
			return;
		}

		const nodeType = nodeTypesStore.getNodeType(node.type, node.typeVersion);
		if (!nodeType) {
			return;
		}

		const connections = mapLegacyConnectionsToCanvasConnections(
			workflowsStore.workflow.connections,
			workflowsStore.workflow.nodes,
		);

		connections.forEach((connection) => {
			const isRelevantConnection = isInput ? connection.target === id : connection.source === id;

			if (isRelevantConnection) {
				const otherNodeId = isInput ? connection.source : connection.target;

				const otherNode = workflowsStore.getNodeById(otherNodeId);
				if (!otherNode || !connection.data) {
					return;
				}

				const [firstNode, secondNode] = isInput ? [otherNode, node] : [node, otherNode];

				if (
					!isConnectionAllowed(
						firstNode,
						secondNode,
						connection.data.source,
						connection.data.target,
					)
				) {
					void nextTick(() => deleteConnection(connection));
				}
			}
		});
	}

	function revalidateNodeInputConnections(id: string) {
		return revalidateNodeConnections(id, 'input');
	}

	function revalidateNodeOutputConnections(id: string) {
		return revalidateNodeConnections(id, 'output');
	}
```

Reply:
Great idea. Applied it.